### PR TITLE
rework for devonfw release 3.2.x

### PR DESCRIPTION
**Pull request by phschimm**
* updated lots of images & text to be more consistent across the board
* removed deprecated links and images
* updated the instructions for project creation to be compatible with the new devonfw IDE

### CODE REVIEW

File: documentation/devon4j-layers.asciidoc [hunk 1]
```diff
@@ -250,52 +244,48 @@ public class BookingmanagementImpl extends AbstractComponentFacade implements Bo
...
+Main benefits of using _Transfer Objects_:
 
 - Avoid inconsistent data (when entities are sent across the app changes tend to take place in multiple places).
 
 - Define how much data to transfer (relations lead to transferring too much data).
 
 - Hide internal details.
 
-In _devon4j_ we can find two different _Transfer Objects:
+In _devon4j_ we can find two different _Transfer Objects_:
```
Comment (dhelmke) [inline documentation/devon4j-layers.asciidoc]: Are there not three TO? Or is the 'BusinessObject' not part of this as well? At least the guide-pdf says so.

File: documentation/devon4j-overview.asciidoc [hunk 2]
```diff
@@ -49,25 +49,22 @@ For _persistence_ and _data access_ devon4j implements:
...
 * http://cxf.apache.org/[Apache CXF] that provides support for:
 
-** REST services through https://en.wikipedia.org/wiki/Java_API_for_RESTful_Web_Services[JAX-RS]
+** https://en.wikipedia.org/wiki/Representational_state_transfer[REST] services through https://en.wikipedia.org/wiki/Java_API_for_RESTful_Web_Services[JAX-RS]
 
-** SOAP services through https://en.wikipedia.org/wiki/Java_API_for_XML_Web_Services[JAX-WS]
+** https://en.wikipedia.org/wiki/SOAP[SOAP] services through https://en.wikipedia.org/wiki/Java_API_for_XML_Web_Services[JAX-WS]
 
 == devon4j Tools
+The following tools are included in the https://github.com/devonfw/ide[devon IDE], which -- mong many other tasks -- can create, run or deploy devon4j applications, avoiding lots of manual work.
```
Comment (dhelmke) [inline documentation/devon4j-overview.asciidoc]: among, not mong I guess?

**Comment by phschimm** [comment]
@dhelmke Thank you for your keen eye! These were easy to miss.

File: documentation/devon4j-layers.asciidoc (hunk 1 above)
Comment (phschimm) [reply documentation/devon4j-layers.asciidoc]: I guess you're right, but the documentation only mentions BusinessObject's in the context of the naming scheme, i.e.: `«BusinessObject»Eto` and `«BusinessObject»«Subset»Cto`. There's also a "Service Transfer Object" mentioned in the guide (which is supposed to follow the naming scheme `«BusinessObject»Sto`) which isn't referenced here either ...

@sjimenez77 do you have any input on this?

**Pull request merged**